pkg/chunked/compression_linux.go

```diff
@@ -234,6 +243,121 @@ func readZstdChunkedManifest(blobStream ImageSourceSeekable, tocDigest digest.Di
 	return decodedBlob, toc, decodedTarSplit, int64(manifestChunk.Offset), err
 }
 
+// ensureTOCMatchesTarSplit validates that toc and tarSplit contain _exactly_ the same entries.
+func ensureTOCMatchesTarSplit(toc *internal.TOC, tarSplit []byte) error {
+	pendingFiles := map[string]*internal.FileMetadata{} // Name -> an entry in toc.Entries
+	for i := range toc.Entries {
+		e := &toc.Entries[i]
+		if e.Type != internal.TypeChunk {
```

Comment:
Very minor but I paused on this for a bit, uncertain; perhaps:

```
// Chunks are just part of files, they won't appear explicitly
// in the tar stream, so we don't validate them.
if e.Type == internal.TypeChunk {
    continue
}
```
?

Reply:
I think this would be best documented somewhere around `pkg/chunked/internal/compression.go`, this is “just” straightforwardly consuming the structure as designed.

https://github.com/containers/storage/pull/1939 did add the basic documentation of `TypeChunk`, although the full semantics of `Offset`/`ChunkOffset` etc. is not written down anywhere.

Reply:
Just to be a good citizen, I have added a commit documenting the format (to the extend I can reverse-engineer it from the code) to the documentation of `internal.FileMetadata`.